zio-http/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -187,6 +190,200 @@ final case class Endpoint[Input, Err, Output, Middleware <: EndpointMiddleware](
+      (err2: Err2) =>
+        err2 match {
+          case sub1: Sub1 => Right(Left(Left(Left(Left(sub1)))))
+          case sub2: Sub2 => Right(Left(Left(Left(Right(sub2)))))
+          case sub3: Sub3 => Right(Left(Left(Right(sub3))))
+          case sub4: Sub4 => Right(Left(Right(sub4)))
+          case sub5: Sub5 => Right(Right(sub5))
+          case _          => Left(s"Unexpected error type")
+        },
+    )
+    copy[Input, alt.Out, Output, Middleware](error = self.error | codec)
+  }
+
+  def outErrors[
+    Err2,
+    Sub1 <: Err2: ClassTag,
+    Sub2 <: Err2: ClassTag,
+    Sub3 <: Err2: ClassTag,
+    Sub4 <: Err2: ClassTag,
+    Sub5 <: Err2: ClassTag,
+    Sub6 <: Err2: ClassTag,
+  ](
+    codec1: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub1],
+    codec2: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub2],
+    codec3: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub3],
+    codec4: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub4],
+    codec5: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub5],
+    codec6: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub6],
+  )(implicit alt: Alternator[Err, Err2]): Endpoint[Input, alt.Out, Output, Middleware] = {
+    val codec = (codec1 | codec2 | codec3 | codec4 | codec5 | codec6).transformOrFail(
+      either => Right(either.left.map(_.left.map(_.left.map(_.left.map(_.merge).merge).merge).merge).merge),
+      (err2: Err2) =>
+        err2 match {
+          case sub1: Sub1 => Right(Left(Left(Left(Left(Left(sub1))))))
+          case sub2: Sub2 => Right(Left(Left(Left(Left(Right(sub2))))))
+          case sub3: Sub3 => Right(Left(Left(Left(Right(sub3)))))
+          case sub4: Sub4 => Right(Left(Left(Right(sub4))))
+          case sub5: Sub5 => Right(Left(Right(sub5)))
+          case sub6: Sub6 => Right(Right(sub6))
+          case _          => Left(s"Unexpected error type")
+        },
+    )
+    copy[Input, alt.Out, Output, Middleware](error = self.error | codec)
+  }
+
+  def outErrors[
+    Err2,
+    Sub1 <: Err2: ClassTag,
+    Sub2 <: Err2: ClassTag,
+    Sub3 <: Err2: ClassTag,
+    Sub4 <: Err2: ClassTag,
+    Sub5 <: Err2: ClassTag,
+    Sub6 <: Err2: ClassTag,
+    Sub7 <: Err2: ClassTag,
+  ](
+    codec1: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub1],
+    codec2: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub2],
+    codec3: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub3],
+    codec4: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub4],
+    codec5: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub5],
+    codec6: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub6],
+    codec7: HttpCodec[HttpCodecType.Status & HttpCodecType.Content, Sub7],
+  )(implicit alt: Alternator[Err, Err2]): Endpoint[Input, alt.Out, Output, Middleware] = {
+    val codec = (codec1 | codec2 | codec3 | codec4 | codec5 | codec6 | codec7).transformOrFail(
```

Comment:
These look fantastic! However, can you pull this functionality out into `HttpCodec.enum(.., ..)`?

Then these implementations will be trivially delegating to `HttpCodec.enum`, and other code can use the feature for various sum types.